conans/client/toolchain/msbuild.py

```diff
@@ -0,0 +1,99 @@
+import os
+import textwrap
+from xml.dom import minidom
+
+from conans.client.tools import msvs_toolset
+from conans.util.files import save, load
+
+
+class MSBuildToolchain(object):
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.definitions = {}
+
+    @ staticmethod
+    def _name_condition(settings):
+        toolset = msvs_toolset(settings)
+
+        props = [("Configuration", settings.build_type),
+                 # FIXME: This probably requires mapping ARM architectures
+                 ("Platform", {'x86': 'Win32',
+                               'x86_64': 'x64'}.get(settings.get_safe("arch"))),
+                 ("PlatformToolset", toolset)]
+
+        name = "".join("_%s" % v for _, v in props if v is not None)
+        condition = " And ".join("'$(%s)' == '%s'" % (k, v) for k, v in props if v is not None)
+        return name.lower(), condition
+
+    def write_toolchain_files(self):
+        name, condition = self._name_condition(self._conanfile.settings)
+        config_filename = "conan_toolchain{}.props".format(name)
+        self._write_config_toolchain(config_filename)
+        self._write_main_toolchain(config_filename, condition)
+
+    def _write_config_toolchain(self, config_filename):
+
+        def format_macro(k, value):
+            return '%s="%s"' % (k, value) if value is not None else k
+
+        runtime = self._conanfile.settings.get_safe("compiler.runtime")
```

Comment:
I think we should respect `compiler.base` (for Intel and other base toolchains).
maybe also introduce `runtime_from_settings`? (like `cppstd_from_settings`)